### PR TITLE
Fixed isPlayerHudComponentVisible not working with showhud

### LINKED ISSUE #547: isPlayerHudComponentVisible only works with setPlayerHudComponentVisible
**Describe the bug**
when Using isPlayerHudComponentVisible( string Component) returns always true until using setPlayerHudComponentVisible

Examples when it is wrong:
1. User uses showHud command there is no Hud but isPlayerHudComponentVisible says there is a hud visible
2. the breath is always true but when a player is not in water there is no breath in the hud
3. sing Camerafunctions like setCameraTarget oder setCameraMatrix always hide Health, Money, Weapon... but isPlayerHudComponentVisible says it is visible.

**To reproduce**
- `showhud`
- `crun isPlayerHudComponentVisible("health")` returns true when should be false

**Expected behaviour**
To return false when component not on screen.

**Screenshots**

**MTA Client:** 
 - Version: 
 - Running in Wine? 

**MTA Server:** 
 - Version 
 - OS: 

**Additional context**
Add any other context about the problem here

From https://bugs.mtasa.com/view.php?id=7356

### PR DESCRIPTION
**Pull request by TracerDS**
Resolves #547

### CODE REVIEW

File: Client/game_sa/CHudSA.cpp [hunk 1]
```diff
@@ -43,6 +43,8 @@ void CHudSA::Disable(bool bDisabled)
     else
         MemPut<BYTE>(FUNC_Draw, 0x80);
 
+    SetComponentVisible(HUD_ALL, !bDisabled);
```
Comment (TheNormalnij) [inline Client/game_sa/CHudSA.cpp]: Does it work when HUD is partially hidden?
1. Disable 'money' component
2. showhud -> all hud is disabled
3. showhud again -> ?

Comment (TracerDS) [reply Client/game_sa/CHudSA.cpp]: it will show all components. Will fix that

File: Client/game_sa/CHudSA.cpp [hunk 2]
```diff
@@ -43,6 +43,9 @@ void CHudSA::Disable(bool bDisabled)
     else
         MemPut<BYTE>(FUNC_Draw, 0x80);
 
+    for (const auto& component : m_HudComponentMap)
+        SetComponentVisible(component.first, !IsComponentVisible(component.first));
```
Comment (TheNormalnij) [inline Client/game_sa/CHudSA.cpp:47]: It looks like `isPlayerHudComponentVisible` will return true for a hidden component after `showhud`

Comment (TracerDS) [reply Client/game_sa/CHudSA.cpp:47]: Hmm, do you think a copy of the components map would be good here?
To save status of each component and then restore it on the next call

**Comment by TheNormalnij** [comment]
Check conversation  in #547.
We need a way to know why a component is hidden.

@qaisjp suggested 
```lua
bool isPlayerHudComponentVisible(string component, bool checkEnabled=true)
```

I'd like to keep `isPlayerHudComponentVisible`  only for `setPlayerHudComponentVisible`.
A new `isHudEnabled` function can be added instead. 
```lua
-- returns true by default and false when showhud command is used
bool isHudEnabled()
```


Both variants are valid.

**Comment by FileEX** [comment]
Any update?

**Comment by TracerDS** [comment]
Im not interested in advancing this PR further anymore.
If someone wants to refresh it, feel free to do so.

**Pull request closed without merging**